app/scripts/controllers/reviewRegistration.js

```diff
@@ -68,98 +68,55 @@ angular.module('confRegistrationWebApp')
       return _.find($scope.blocks, {id: blockId});
     };
 
-    $scope.confirmRegistration = function () {
-      $scope.submittingRegistration = true;
-
-      /*if the totalPaid (previously) AND the amount of this payment are less than the minimum required deposit, then
-        show and error message. the first payment must be at least the minimum deposit amount.  subsequent payments
-        can be less than the amount.  this is confirmed by making sure the total previously paid is above the min deposit amount.
-        */
-      if ($scope.currentRegistration.pastPayments.length === 0 && Number($scope.currentPayment.amount) < $scope.currentRegistration.calculatedMinimumDeposit) {
-        $scope.currentPayment.errors.push('You are required to pay at least the minimum deposit of ' + $filter('currency')(registration.calculatedMinimumDeposit, '$') + ' to register for this event.');
+    $scope.getConfirmButtonName = function () {
+      if (currentRegistration.completed || !$scope.spouseRegistration) {
+        return gettextCatalog.getString('Confirm');
+      } else {
+        return gettextCatalog.getString('one of the Register buttons');
       }
+    };
 
-      if(Number($scope.currentPayment.amount) > $scope.currentRegistration.remainingBalance) {
-        $scope.currentPayment.errors.push('You are paying more than the total due of ' + $filter('currency')(registration.remainingBalance, '$') + ' to register for this event.');
-      }
-      if (Number($scope.currentPayment.amount) === 0 || !$scope.acceptedPaymentMethods()) {
-        setRegistrationAsCompleted();
-        return;
-      }
+    // Return a boolean indicating whether the register button(s) should be disabled
+    $scope.registerDisabled = function () {
+      return $scope.registerMode === 'preview' || !$scope.allRegistrantsValid() || $scope.submittingRegistration;
+    };
 
-      if (!_.isEmpty($scope.currentPayment.errors)) {
-        modalMessage.error({
-          'title': 'Please correct the following errors:',
-          'message': $scope.currentPayment.errors
-        });
```

Comment:
I think I've usually just seen the [`gettext`](https://angular-gettext.rocketeer.be/dev-guide/annotate-js/) service used like `var myString = gettext("Hello");`. Are there benefits to using the catalog service? I'm not the expert.